### PR TITLE
47013 Client for Travel Reimbursement Auth Endpoint

### PR DESCRIPTION
**Pull request by randomsync**
## Description of change

This PR adds a client to get the token from auth endpoint for travel reimbursement api.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#47013

## Things to know about this PR

The settings are added under `check_in` key for rspecs. Settings for other environments are being handled as a part of another ticket.

- [x] rspec
- [x] rubocop

### CODE REVIEW

File: config/settings/test.yml [hunk 1]
```diff
@@ -204,6 +204,13 @@ sidekiq:
   github_oauth_secret: 000xxx
 
 check_in:
+  travel_reimbursement_api:
+    auth_url: https://login.microsoftonline.us
```
Comment (kanchanasuriya) [inline config/settings/test.yml:208]: Do you think we can have a separate section for auth & claims api?

File: modules/check_in/app/services/travel_claim/client.rb [hunk 2]
```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+module TravelClaim
+  ##
```
Comment (kanchanasuriya) [inline modules/check_in/app/services/travel_claim/client.rb:4]: Wondering if we can have one more sub-module defined for VEIS under travel_claim?  That way when we create a client interface for claims service under the claims sub-module and all BTSSS clients under travel_claim.

File: config/settings/test.yml [hunk 3]
```diff
@@ -204,6 +204,13 @@ sidekiq:
...
+    tenant_id: 6d4d703a-2d9a-4c2a-ac90-976e52eacb52
+    client_id: 9f124d0c-e3d0-4a02-a540-64def7ef3889
+    client_secret: f3b721aa-f9c1-4ff2-bf47-9b3a30757013
```
Comment (holdenhinkle) [inline config/settings/test.yml:211]: Can you please confirm that these settings are not real?

**Review by RachalCassity — DISMISSED** [review]

File: config/settings/test.yml [hunk 4]
```diff
@@ -204,6 +204,13 @@ sidekiq:
   github_oauth_secret: 000xxx
 
 check_in:
```
Comment (holdenhinkle) [inline config/settings/test.yml:206]: Are any of these settings not used for tests? If so, you'll have to create a corresponding [devops](https://github.com/department-of-veterans-affairs/devops/tree/master/ansible/deployment/config/vets-api) PR for the `settings.yml` for each environment.

**Review by holdenhinkle — COMMENTED** [review]
I made a few comments. Thanks!

File: config/settings/test.yml (hunk 3 above)
Comment (randomsync) [reply config/settings/test.yml:211]: yes, I can confirm. these are randomly generated uuids for rspec purposes.

File: config/settings/test.yml (hunk 4 above)
Comment (randomsync) [reply config/settings/test.yml:206]: Yes, we are handling adding configs in devops repo in a separate ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/47014

File: config/settings/test.yml (hunk 1 above)
Comment (randomsync) [reply config/settings/test.yml:208]: My thinking behind this was to keep the settings and code for claim submission in one place.

File: modules/check_in/app/services/travel_claim/client.rb (hunk 2 above)
Comment (randomsync) [reply modules/check_in/app/services/travel_claim/client.rb:4]: As we discussed, I think it'll be easier to maintain if we have the token endpoint in the same client/service object as the claims submission.

**Review by holdenhinkle — APPROVED** [review]

**Review by kanchanasuriya — APPROVED** [review]
LGTM

**Pull request merged**